src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1284,10 +1279,12 @@ private void ImportCall(ILOpcode opcode, int token)
                 }
             }
 
-            HandleCall(callee, callee.Signature, opcode);
+            TypeDesc localConstrainedType = _constrainedType;
+            _constrainedType = null;
```

Comment:
Nit: might be better to clear `_constrainedType` in `EndImportingInstruction` so that a constrained prefix in an invalid location doesn't apply to the next `call`.

Reply:
Good idea, done.

Reply:
Maybe after that you no longer need to clear it here and pass it around as an extra parameter?

Reply:
We can end up in HandleCall from several different points in handling an instruction (to do things like call into runtime helpers), so clearing it out early can prevent any future accidental constraints.